initialization/snyk.sh

```diff
@@ -1,19 +1,45 @@
+#!/bin/bash
+
+[ -z "$SNYK_TOKEN" ] && { echo "Set 'context: snyk' in your .circleci/config.yml workflow"; exit 1; }
+
 LATEST_RELEASE=$(curl -L -s -H 'Accept: application/json' https://github.com/snyk/snyk/releases/latest)
 LATEST_VERSION=$(echo $LATEST_RELEASE | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
 BINARY_URL="https://github.com/snyk/snyk/releases/download/$LATEST_VERSION/snyk-linux"
 
-# [ -z "$SNYK_LEVEL" ] && { echo "Set SNYK_LEVEL = 'RPRT' in the CircleCI environment variables settings"; exit 1; }                                                           
-[ -z "$SNYK_TOKEN" ] && { echo "Set 'context: snyk' in your .circleci/config.yml workflow"; exit 1; }
+curl -sL "$BINARY_URL" -o snyk
+[ ! -f snyk ] && { echo "Snyk failed to download!"; exit 0; }
+chmod +x snyk
 
-curl -sL "$BINARY_URL" -o snyk-linux
+severity_threshold="${severity_threshold:-low}" # by default show all vulns
+fail_on="${fail_on:-never}" # by default never fail (backwards compatibility)
+debug="${debug:-false}" # debug output is messy
 
-[ ! -f snyk-linux ] && { echo "Snyk failed to download!"; exit 0; }
+if [[ "${fail_on}" = "never" ]]; then
+  fail_on=all
+  NEVER_FAIL=true 
+fi
+
+flags=(
+  "--severity-threshold=$severity_threshold"
+  "--fail-on=$fail_on"
+  "--org=$org"
+)
+
+monitor_flags=(
+  "--org=$org"
+)
+
+if [[ $debug =~ (true|on|1) ]] ; then
+  flags+=( "-d" )
+  monitor_flags+=( "-d" )
+fi
 
-chmod +x snyk-linux
+# suppresses errors w/ snyk monitor (which shouldn't have any)
+./snyk monitor ${monitor_flags[@]} || true
 
-./snyk-linux test -d || true # will always pass, but still send results up to Snyk                                                                                                
-./snyk-linux monitor -d || true # suppresses errors w/ snyk monitor (which shouldn't have any)                                                                                    
+echo "~~~ Running Snyk tests"
+./snyk test ${flags[@]}
 
-if [ "$SNYK_LEVEL" = "FLHI" ]; then
-    ./snyk-linux test --severity-threshold=high -q
+if [[ "${NEVER_FAIL}" = true ]]; then
+  exit 0;
```

Comment:
This is based on https://github.com/segmentio/snyk-buildkite-plugin/blob/master/hooks/command#L73-L75

I thought it needed to be `true`, not `"true"` and that `set +e` would just maintain the existing failure code, not guarantee the code would be 0 so made these changes. Does the BK version need to get updated?